plugins/SliceInfoPlugin/SliceInfo.py

```diff
@@ -24,6 +21,33 @@
 
 catalog = i18nCatalog("cura")
 
+class SliceInfoThread(Thread):
```

Comment:
Instead of using Thread directly, why not create a Job subclass for this? Those are specifically designed for this.